servers/rendering/renderer_scene_cull.h

```diff
@@ -109,6 +110,84 @@ class RendererSceneCull : public RendererScene {
 	virtual void camera_set_use_vertical_aspect(RID p_camera, bool p_enable);
 	virtual bool is_camera(RID p_camera) const;
 
+	/* OCCLUDER API */
+
+	virtual RID occluder_allocate();
+	virtual void occluder_initialize(RID p_occluder);
+	virtual void occluder_set_mesh(RID p_occluder, const PackedVector3Array &p_vertices, const PackedInt32Array &p_indices);
+
+	class RendererSceneOcclusionCull {
```

Comment:
I would just make all of this inline, not virtual. I dont think we will have more than one implementation. If we do then we see how to do it, but for now I think its not worth it.

Reply:
Right now those are virtual because there is the "dummy" default implementation and the actual working implementation is loaded from the module. I can change it to just have a pointer and check whether it's null or not before all the calls, other than that I don't know how to make it an optional thing.

Reply:
Ah I see better now how this works. IMO I would make the HZB code a bit more encapsulated, and would make the `is_occluded()` function _FORCE_INLINE_ for performance. Also you can probably move these classes out to their own header/cpp to make it cleaner.